src/common/components/ui/formik/SwitchField.tsx

```diff
@@ -14,13 +14,34 @@ const SwitchField: React.FC<SwitchFieldProps> = ({
   getHelperText,
   idPostfix,
   labelIcon,
+  tooltipProps,
   ...props
 }) => {
   const [field, { touched, error }] = useField(props.name);
   const fieldId = getFieldId(props.name, 'input', idPostfix);
   const isValid = !(touched && error);
   const errorMessage = !isValid ? error : '';
   const hText = getHelperText ? getHelperText(field.value) : helperText;
+  let switchElement = (
+    <Switch
+      {...field}
+      id={fieldId}
+      label={label}
+      isDisabled={props.isDisabled}
+      isChecked={field.value}
+      onChange={(checked, event) => {
+        if (onChangeCustomOverride) {
+          onChangeCustomOverride(checked, event);
+        } else {
+          field.onChange(event);
+          onChange && onChange(checked, event);
+        }
+      }}
+    />
+  );
+  if (tooltipProps) {
+    switchElement = <Tooltip {...tooltipProps}>{switchElement}</Tooltip>;
```

Comment:
This can be made a reusable component.

Example:
```
// Step 1: Define a HOC
const withTooltip = (SomeComp, tooltipProps) => (someCompProps) => (
  <Tooltip {...tooltipProps}>
    <SomeComp {...someCompProps} />
  </Tooltip>
);

// Step 2: Prepare the Switch props
const switchProps = {
  ...field,
  id: fieldId,
  label,
  isDisabled: props.isDisabled,
  isChecked: field.value,
  onChange: (checked, event) => {
    if (onChangeCustomOverride) {
      onChangeCustomOverride(checked, event);
    } else {
      field.onChange(event);
      onChange && onChange(checked, event);
    }
  },
};

/////////////BEFORE///////////////////////////////////
// let switchElement = (
//   <Switch
//     {...field}
//     id={fieldId}
//     label={label}
//     isDisabled={props.isDisabled}
//     isChecked={field.value}
//     onChange={(checked, event) => {
//       if (onChangeCustomOverride) {
//         onChangeCustomOverride(checked, event);
//       } else {
//         field.onChange(event);
//         onChange && onChange(checked, event);
//       }
//     }}
//   />
// );
/////////////BEFORE/////////////////////////////////// 

// Step 3: Decide which Switch should render
let switchElement = <Switch {...switchProps} />
if (tooltipProps) {
  const SwitchWithTooltip = withTooltip(Switch, tooltipProps);
  switchElement = <SwitchWithTooltip {...switchProps} />;
}
```
Extra: [working example](https://codesandbox.io/embed/magical-snyder-k1wx2?fontsize=14&hidenavigation=1&theme=dark)